addons/mail/static/tests/thread/thread.test.js

```diff
@@ -457,51 +443,6 @@ test("show empty placeholder when thread contains no message", async () => {
     await contains(".o-mail-Message", { count: 0 });
 });
 
-test("show empty placeholder when thread contains only empty messages", async () => {
-    const pyEnv = await startServer();
-    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
-    pyEnv["mail.message"].create({ model: "discuss.channel", res_id: channelId });
-    await start();
-    await openDiscuss(channelId);
-    await contains(".o-mail-Thread", { text: "The conversation is empty." });
-    await contains(".o-mail-Message", { count: 0 });
-});
-
-test("message list with a full page of empty messages should load more messages until there are some non-empty", async () => {
```

Comment:
This test in particular looks ok to remove fully.
I haven't seen in code if we could get rid of the code that was handling that?

Reply:
The code is basically `messages` in `Thread` component that under-the-hood was filtering empty messages, which in the end resulted in `message.length === 0` for the UI showing. With the change I made we just look at actual `thread.messages` including user-deleted one. Code that could be removed in message_model was removed.